src-v5/carousel.tsx

```diff
@@ -171,14 +171,14 @@ export const Carousel = (props: CarouselProps): React.ReactElement => {
     }
   }, [currentSlide, moveSlide, propsSlidesToScroll, scrollMode, wrapAround]);
 
-  // Why is this needed?
-  // useEffect(() => {
-  //   if (typeof slideIndex === 'number' && !autoplayReverse) {
-  //     moveSlide(slideIndex);
-  //   }
-  // }, [slideIndex, autoplayReverse, moveSlide]);
-
-  // Why is this needed?
+  // When user changed the slideIndex property from outside.
+  useEffect(() => {
+    if (typeof slideIndex === 'number' && !autoplayReverse) {
+      moveSlide(slideIndex);
+    }
+  }, [slideIndex, autoplayReverse]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
moveSlide is changed when some of the navigation buttons are used and cause triggering this hook. That's why we don't need it in the dependencies